### ISSUE TITLE
[Project 2] Test Case 4_4

### ISSUE THREAD

**Issue by domlimm** [nus-cs2030/2021-s1#457]
Hi, i think this is the worst error that has literally no solution or hopes to debug, but has anyone come across it?

Appreciate any suggestions... I've been on this level for 5 days......... 

The start of it:

![image](https://user-images.githubusercontent.com/46486515/99802239-4e9b5180-2b72-11eb-93fc-8f2b7d81e796.png)

The end of it:

![image](https://user-images.githubusercontent.com/46486515/99802270-61ae2180-2b72-11eb-80aa-5dabac79b05d.png)

**Comment by jeremybrett4869** [comment]
For debugging, maybe you can try printing out the queue of the server to check whether it is updated properly?

**Comment by aaron-ljx** [comment]
If you're using an IDE like IntelliJ, you can try to see what is the current state of your shop at 3.854 @ 31 arrives and how the shop changes into a new state as you execute that arrive event. There is probably an edge case that your execute function didn't handle properly.

**Comment by xingaporean** [comment]
the order of scheduling is the problem or u not using the most updated shop or u have a small bug some where when u create ur events

**Comment by Mavvvvv** [comment]
I will recommend you to print out each size of the queue in every server. From there, visually check if the queue is full.
You can trace the size of the queue and figure out what went wrong.

you can add something like this to your ArrivalEvent: System.out.println(shop.getSize());
and this to your Shop:    public List<Integer> getSize(){
        List<Integer> sizeList = this.serverList.stream().map(x -> x.getCustList().size()).collect(Collectors.toList());
        return sizeList;
    }

**Comment by TypeDefinition** [comment]
If you want to get the input from the test case, do something like

```
if (output.equals("0.123 1 served by server 2")) {
    printOutput();
}
```

I also recommend you print out all the states of the servers before and after every single event in order to trace what went wrong.

So something like:

Before Event: 
Server 1: Available: false, Next Available Time: 3.123, Queue: 2
Wait Event Occurs
After Event:
Server 1: Available: false, Next Available Time: 3.123, Queue: 3

**Comment by domlimm** [comment]
Thank you all for your comments! I had to re-design my program, thank you so much yall!
@jeremybrett4869 @aaron-ljx @xing221 @Mavvvvv @TypeDefinition 🤟🏻
